hcl2template/addrs/plugin.go

```diff
@@ -13,21 +16,58 @@ import (
 
 // Plugin encapsulates a single plugin type.
 type Plugin struct {
-	Hostname  string
-	Namespace string
-	Type      string
+	Source string
 }
 
-func (p Plugin) RealRelativePath() string {
-	return p.Namespace + "/packer-plugin-" + p.Type
+// Parts returns the list of components of the source URL, starting with the
+// host, and ending with the name of the plugin (non-simplified).
+//
+// This will correspond more or less to the filesystem hierarchy where
+// the plugin is installed.
+func (p Plugin) Parts() []string {
+	var retParts []string
+
+	// This should not happen, we're supposed to check that the source isn't
+	// empty when creating the plugin, but let's be careful here and not
+	// misbehave.
+	if p.Source == "" {
+		log.Printf("[ERROR] - Plugin.Parts: the source was empty, this should not happen in normal usage, and is likely a Packer bug.")
+		return retParts
+	}
+
+	dir, part := path.Split(strings.TrimRight(p.Source, "/"))
+	for dir != "" {
+		retParts = append(retParts, part)
+		dir, part = path.Split(strings.TrimRight(dir, "/"))
+	}
+	retParts = append(retParts, part)
+
+	// The algorithm above registers the path in reversed order
+	//
+	// The function is supposed to return them in the host->...->name
+	// order however, so we have to reverse the slice.
+	// There isn't (afaik) a standard function for that in Go, so we write
+	// it manually.
+	reversedParts := make([]string, len(retParts))
```

Comment:
Seeing as Packer is using Go 1.21. The slices pkg has a Reverse function you can use here.